### PR TITLE
Fail fast when image is specified without tag

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
~~Currently when replacing images we assume an image is in the form `image:tag` but there are cases where this is not true. This PR defaults tag to `:latest` when not specified.~~
This PR changes image replace code such that it requires `image:tag` format. 

#### Special notes for your reviewer:
/assign @smarterclayton 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -373,10 +373,16 @@ func ReplaceRegistryInImageURL(imageURL string) (string, error) {
 			}
 		}
 		last := strings.SplitN(parts[countParts-1], ":", 2)
+		var version string
+		if len(last) == 1 {
+			version = "latest"
```
Comment (smarterclayton) [inline test/utils/image/manifest.go:378]: Can we not just fix this and make the error be fail fast?    It seems wierd that we'd allow pull specs without tags anywhere.

File: test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml [hunk 2]
```diff
@@ -73,7 +73,7 @@ spec:
         # test for directories or create them. It needs additional privileges
         # for that.
         - name: busybox
-          image: k8s.gcr.io/busybox
+          image: k8s.gcr.io/e2e-test-images/busybox:1.29-1
```
Comment (smarterclayton) [inline test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml:76]: Yay, we caught a bug!

Comment (dims) [reply test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml:76]: now do we need a test for a test util LOL (kidding)

Comment (soltysh) [reply test/e2e/testing-manifests/storage-csi/mock/csi-mock-proxy.yaml:76]: lol, I was considering that :stuck_out_tongue_winking_eye:

**Pull request merged**